devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -1365,6 +1866,579 @@ def deleteCode(codeID) {
+		def cmds = []
+		
+		def codeID = Integer.parseInt(schSettings.codeId, 10)
+		
+		def startDate = schSettings.startDate
+		def startTime = schSettings.startTime
+		def endDate = schSettings.endDate
+		def endTime = schSettings.endTime
+		
+		state["setYearDayStartDate$codeID"] = startDate
+		state["setYearDayStartTime$codeID"] = startTime
+		state["setYearDayStopDate$codeID"] = endDate
+		state["setYearDayStopTime$codeID"] = endTime
+		
+		def startDateTime = getYearDayTimeDetails(startDate, startTime)
+		def startHour = startDateTime.hour
+		def startMinute = startDateTime.minute
+		def startDay = startDateTime.day
+		def startMonth = startDateTime.month
+		def startYear = startDateTime.year.toString()
+		startYear = startYear.substring(2, startYear.length())
+		startYear = Integer.parseInt(startYear, 10)
+		
+		def endDateTime = getYearDayTimeDetails(endDate, endTime)
+		def stopHour = endDateTime.hour
+		def stopMinute = endDateTime.minute
+		def stopDay = endDateTime.day
+		def stopMonth = endDateTime.month
+		def stopYear = endDateTime.year.toString()
+		stopYear = stopYear.substring(2, stopYear.length())
+		stopYear = Integer.parseInt(stopYear, 10)
+		
+		cmds << secure(zwave.scheduleEntryLockV3.scheduleEntryLockYearDaySet(scheduleSlotId: 1, setAction: 1, startDay: startDay, startHour: startHour, startMinute: startMinute,
+				startMonth: startMonth, startYear: startYear, stopDay: stopDay, stopHour: stopHour, stopMinute: stopMinute, stopMonth: stopMonth,
+				stopYear: stopYear, userIdentifier: codeID))
+		cmds << requestYearDaySchedule(codeID)
+		cmds = delayBetween(cmds, 200)
+	   
+	   log.debug "[DTH] getYearDayScheduleSetCmd() returning with commands := $cmds"
+	   return cmds
+	}
+	return null
+}
+
+/**
+ * Returns the command(s) for deleting week day schedule
+ * 
+ * @param codeID: The code slot number
+ *
+ * @return The command(s) for deleting week day schedule
+ */
+private def getDailyRepeatingDeleteScheduleCmd(codeID) {
+	log.trace "[DTH] Executing 'getDailyRepeatingDeleteScheduleCmd()' by ${device.displayName} with codeID := $codeID"
+	def cmds = []
+	cmds << secure(zwave.scheduleEntryLockV3.scheduleEntryLockDailyRepeatingSet(scheduleSlotId: 1, setAction: 0, userIdentifier: codeID))
+	cmds << requestDailyRepeatingSchedule(codeID)
+	cmds = delayBetween(cmds, 200)
+
+	log.debug "[DTH] 'getDailyRepeatingDeleteScheduleCmd()' returning with commands := $cmds"
+	cmds
+}
+
+/**
+ * Returns the command(s) for deleting week day schedule
+ * 
+ * @param codeID: The code slot number
+ *
+ * @return The command(s) for deleting week day schedule
+ */
+private def getWeekDayDeleteScheduleCmd(codeID) {
+	def cmds = []
+	def setCmds = []
+	def getCmds = []
+	for (def i = 0; i <= 6; i++) {
+		setCmds << secure(zwave.scheduleEntryLockV3.scheduleEntryLockWeekDaySet(setAction: 0, scheduleSlotId: (i + 1), userIdentifier: codeID))
+		getCmds << requestWeekDaySchedule((i + 1), codeID)
+	}
+	if (setCmds && getCmds) {
```

Comment:
Is there any way both of these won't be populated?

Reply:
Also doesn't the `requestAllWeekDaySchedule` function below generate `getCmds` exactly?